lib/src/test/kotlin/integration/account/AccountTest.kt

```diff
@@ -249,6 +249,7 @@ class AccountTest {
         // If this test starts randomly falling, try recalculating class hash.
         // 2. If it fails on CI, make sure to delete the compiled contracts before running this test.
         // Chances are, the contract was compiled with a different compiler version.
+        // 3. This test sometimes fails due to getNonce receiving higher (pending) nonce than addDeclareTransaction expects
```

Comment:
is there any way we could circumvent this?

Reply:
Only by adding big sleep times, preventing nonce-changing tests from parallel running, separating gateway and RPC tests from each other, also adding sleep gaps between those. 

Since those are integration tests and meant for local runs for now, I wouldn't go into it. It would make more sense to try that after gateway removal, but I'm not sure it's worse the effort, we should have devent-rs by then and probably still won't rely on integration tests for regular and CI runs since their inconsistency.